tests/compiler/tnimblecmd.nim

```diff
@@ -0,0 +1,26 @@
+include compiler/[nimblecmd]
```

Comment:
@xflywind how about:
```nim
import compiler/[nimblecmd,options,lineinfos]
import std/[strtabs,sequtils]
```

```
# in nimblecmd.nim:
when defined(testing):
  export chosen, addPackage
```

note that {.privateImport.} (https://github.com/nim-lang/Nim/pull/11865) would be much better (no global effect on other modules) but until it's merged this is still the better option, compared to include, which has many issues (as explained in #11865)

Reply:
I will check it.

Reply:
The only reason why ``.privateImport`` doesn't have "many issues" is because almost nobody used it.

Reply:
> The only reason why .privateImport doesn't have "many issues" is because almost nobody used it.

@araq the reason nobody used it except myself is because the PR still hasn't been merged. 
I've rebased it recently to address bitrot conflicts with devel so it should still be good to go / ready to address any further review comments.
The safeguards are in place:
not only you can do `--experimental:allowPrivateImport`, but it can also be pushed/popped for more targetted use:
```nim
# in tnimblecmd
{.push experimental: "allowPrivateImport".}
from nimblecmd {.privateImport.} import chosen, addPackage
...
{.pop.}
```
=> other modules importing nimblecmd won't see `chosen`, `addPackage`, and `nimblecmd` won't be 
duplicated, breaking modularity and other common issues with include